File: Data/Language/language.py
import json

class Language:
    def __init__(self):
        # Default language setting
        self.default_language = 'en'  # Assuming English is the default language
        # Dictionary to store loaded language translations
        self.translations = {}
        # Load language files
        self.load_language_files()

    # Load language files
    def load_language_files(self):
        # Assuming language files are stored in the same directory
        # and follow a specific naming convention (e.g., en.json, fr.json)
        languages = ['en', 'fr']  # Example languages (English and French)
        for language in languages:
            try:
                with open(f'{language}.json', 'r', encoding='utf-8') as file:
                    # Store translations in the translations dictionary
                    self.translations[language] = json.load(file)
            except FileNotFoundError as e:
                print(f'Failed to load language file for {language}: {e}')

    # Method to get a translated string for a given key and language
    def get_string(self, key, language):
        # Retrieve the translation for the provided key and language
        translation = self.translations.get(language, self.translations.get(self.default_language, {}))
        return translation.get(key, key)  # Return the translated string or the key if not found

    # Method to switch the current language
    def set_language(self, language):
        # Implement logic to switch the current language
        # For simplicity, you can just update a variable storing the current language
        # and reload the UI to reflect the language change
        # You may also want to persist the selected language in local storage or a similar mechanism
        pass

File: Data/Language/language.cs
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

public class Language
{
    // Default language setting
    private string defaultLanguage = "en"; // Assuming English is the default language
    // Dictionary to store loaded language translations
    private Dictionary<string, Dictionary<string, string>> translations = new Dictionary<string, Dictionary<string, string>>();

    public Language()
    {
        // Load language files
        LoadLanguageFiles();
    }

    // Load language files
    private void LoadLanguageFiles()
    {
        // Assuming language files are stored in the same directory
        // and follow a specific naming convention (e.g., en.json, fr.json)
        string[] languages = { "en", "fr" }; // Example languages (English and French)
        foreach (string language in languages)
        {
            try
            {
                string json = File.ReadAllText($"{language}.json");
                // Store translations in the translations dictionary
                translations[language] = JsonConvert.DeserializeObject<Dictionary<string, string>>(json);
            }
            catch (FileNotFoundException e)
            {
                Console.WriteLine($"Failed to load language file for {language}: {e.Message}");
            }
        }
    }

    // Method to get a translated string for a given key and language
    public string GetString(string key, string language)
    {
        // Retrieve the translation for the provided key and language
        if (translations.TryGetValue(language, out Dictionary<string, string> translation))
        {
            if (translation.TryGetValue(key, out string translatedString))
            {
                return translatedString;
            }
        }
        return translations.ContainsKey(defaultLanguage) && translations[defaultLanguage].ContainsKey(key) ? translations[defaultLanguage][key] : key;
    }

    // Method to switch the current language
    public void SetLanguage(string language)
    {
        // Implement logic to switch the current language
        // For simplicity, you can just update a variable storing the current language
        // and reload the UI to reflect the language change
        // You may also want to persist the selected language in local storage or a similar mechanism
    }
}
